lib/constructs/provisioning-sfn-workflow.ts

```diff
@@ -45,18 +46,36 @@ export interface IProvisioningWorkflow {
  * composed to form a workflow for provisioning jobs from SNOW to the CSPs.
  *
  */
-export class ProvisioningWorkflow implements IProvisioningWorkflow {
+export class ProvisioningWorkflow extends Construct implements IProvisioningWorkflow {
   readonly mappedTasks: TasksMap = {};
   readonly workflow: sfn.IChainable;
   readonly logGroup: logs.ILogGroup;
+  readonly provisioningJobsQueue: sqs.IQueue;
+  readonly resultFn: lambdaNodeJs.NodejsFunction;
 
-  constructor(scope: Construct, props: ProvisioningWorkflowProps) {
+  constructor(scope: Construct, id: string, props: ProvisioningWorkflowProps) {
+    super(scope, id);
     const { environmentName } = props;
+
+    this.provisioningJobsQueue = new sqs.Queue(scope, "ProvisioningJobsQueue", {
+      queueName: "ProvisioningJobsQueue",
+    });
+
     // Provisioning State machine functions
     const sampleFn = new lambdaNodeJs.NodejsFunction(scope, "SampleFunction", {
       entry: "api/provision/sample-fn.ts",
       // vpc: props.vpc,
     });
+    this.resultFn = new lambdaNodeJs.NodejsFunction(scope, "ResultFunction", {
+      functionName: "ResultFunction",
```

Comment:
Do we _really_ want to name this? Won't this result in us only being able to build this once?

Reply:
So to expand on this a bit, `functionName` maps to the `FunctionName` CloudFormation property, which specifies the `FunctionName` as part of the `CreateFunction` API call. The name is the unique identifier of the function (as part of the `resourceName` field of its ARN). Because `Invoke` and other Lambda APIs accept the `FunctionName` parameter to uniquely identify a function, it must be unique within a region within an account. The same is true for nearly all resources that allow specifying the physical name (S3 Buckets, SQS Queues, etc) -- the name is a _unique_ identifier.

If this code is left as-is, we will only be able to build our stack once per region per account. This will break our Sandbox account. But even more significantly, it means that we can only create one `ProvisioningWorkflow` construct per stack.

In general, we should strongly avoid specifying physical names for resources as they don't really matter. We should pass references to functions/queues/buckets not their names (if we do need to pass a string, we use its name attribute). Exceptions are made in our IAM stack so that role names are predictable. But anything else will cause issues for devs working in parallel.

Reply:
Ahh thanks - I've been head-down strictly in the unit tests on this so didn't consider that. Really just wanted a clean way to identify that the particular function in question had been created. Ended up specifying the Role instead: 

```
        Role: stack.resolve((provisioningSfn.resultFn.role?.node.defaultChild as CfnRole).attrArn),
```

Reply:
Might be able to use `description` too if we specify one. But yeah, that seems reasonable.